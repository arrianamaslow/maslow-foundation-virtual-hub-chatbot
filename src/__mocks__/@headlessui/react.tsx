/**
 * @jest-environment jsdom
 */

import { DialogProps } from '@headlessui/react'
import React, { ReactNode } from 'react'

type DialogPanelTestProps = {
  children?: ReactNode
}

interface DialogTestProps extends DialogProps {
  children?: ReactNode
}

const DialogPanel: React.FC<DialogPanelTestProps> = ({ children }) => {
  return <div data-testid='mock-dialog-panel-component'>{children}</div>
}

const Dialog: React.FC<DialogTestProps> = ({ open, children }) => {
  return (
    <div data-testid='mock-dialog-component'>
      {open ? 'DIALOGUEOPENED' : 'DIALOGUECLOSED'}
      {children}
    </div>
  )
}

export { Dialog, DialogPanel }
