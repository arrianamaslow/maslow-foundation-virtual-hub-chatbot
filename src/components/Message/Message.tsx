import React from 'react'
import style from './style.module.css'
import Image from 'next/image'
import OpenAI from 'openai'

export type MessageProps = {
  openAiMessage: OpenAI.ChatCompletionMessageParam
}

const Message: React.FC<MessageProps> = ({ openAiMessage }) => {
  const imageSrc =
    openAiMessage.role === 'user' ? '/img/user_icon.svg' : '/img/maslow-logo-colour.svg'
  return (
    <div className={style[`${openAiMessage.role}Div`]}>
      <Image src={imageSrc} alt={openAiMessage.role} width={26} height={54} />
      <div className={style[`${openAiMessage.role}TextBox`]}>
        {openAiMessage.content?.toString()}
      </div>
    </div>
  )
}

export default Message
