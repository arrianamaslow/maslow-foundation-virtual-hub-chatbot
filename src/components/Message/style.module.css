.userTextBox {
  @apply ml-auto p-2 mr-0 rounded-lg bg-gray-300 w-fit md:max-w-[474px] max-w-[238px] self-stretch text-black text-[17px] md:text-[18px] leading-normal not-italic font-normal break-words;
}
.assistantTextBox {
  @apply rounded-lg w-fit md:max-w-[474px] max-w-[238px] self-stretch text-black text-[17px] md:text-[18px] leading-normal not-italic font-normal break-words;
}
.userDiv {
  @apply flex items-end self-end w-fit flex-row-reverse gap-2;
}
.assistantDiv {
  @apply flex w-fit flex-row self-start items-end gap-2;
}
