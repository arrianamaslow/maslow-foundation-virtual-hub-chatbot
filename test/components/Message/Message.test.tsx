/**
 * @jest-environment jsdom
 */

import { render } from '@testing-library/react'
import Message from '@/components/Message/Message'

describe('<Message />', () => {
  const messageText = 'hello'

  it('renders user message correctly and "user" role is passed as param correctly', () => {
    const message = render(<Message openAiMessage={{ role: 'user', content: messageText }} />)
    expect(message).toMatchSnapshot()
  })

  it('renders assistant message correctly and "assistant" role is passed as param correctly', () => {
    const message = render(<Message openAiMessage={{ role: 'assistant', content: messageText }} />)
    expect(message).toMatchSnapshot()
  })

  it('passes text to user message as param correctly', () => {
    const message = render(<Message openAiMessage={{ role: 'user', content: messageText }} />)
    expect(message.getByText(messageText)).not.toBeNull()
  })

  it('passes text to assistant message as param correctly', () => {
    const message = render(<Message openAiMessage={{ role: 'assistant', content: messageText }} />)
    expect(message.getByText(messageText)).not.toBeNull()
  })

  it('renders user and assistant messages with same text differently', () => {
    const userMessage = render(<Message openAiMessage={{ role: 'user', content: messageText }} />)
    const assistantMessage = render(
      <Message openAiMessage={{ role: 'assistant', content: messageText }} />
    )
    expect(userMessage.asFragment()).not.toEqual(assistantMessage.asFragment())
  })
})
