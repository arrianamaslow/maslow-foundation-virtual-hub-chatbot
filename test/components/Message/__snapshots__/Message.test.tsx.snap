// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Message /> renders assistant message correctly and "assistant" role is passed as param correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="assistantDiv"
      >
        <img
          alt="assistant"
          height="54"
          src="/img/maslow-logo-colour.svg"
          width="26"
        />
        <div
          class="assistantTextBox"
        >
          hello
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="assistantDiv"
    >
      <img
        alt="assistant"
        height="54"
        src="/img/maslow-logo-colour.svg"
        width="26"
      />
      <div
        class="assistantTextBox"
      >
        hello
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<Message /> renders user message correctly and "user" role is passed as param correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="userDiv"
      >
        <img
          alt="user"
          height="54"
          src="/img/user_icon.svg"
          width="26"
        />
        <div
          class="userTextBox"
        >
          hello
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="userDiv"
    >
      <img
        alt="user"
        height="54"
        src="/img/user_icon.svg"
        width="26"
      />
      <div
        class="userTextBox"
      >
        hello
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
